### ISSUE TITLE
When attempting to edit a service that has an image pulled from a private dockerhub repo, the UI kicks back to the login screen

### ISSUE THREAD

**Issue by mpsmith** [swarmpit/swarmpit#201]
---------------------------------------------------
BUG REPORT
---------------------------------------------------

**Description**

When I attempt to edit a service that is using an image from our private Dockerhub repo, the web UI kicks back to the login screen.   

When I edit a service that uses a publicly available image on Dockerhub, the UI behaves as expected and allows editing the service.

I tried adding some credentials in the 'dockerhub' section of the UI to see if that would resolve, but it did not seem to have any affect.   Do I need a dockerhub credential there for this to function?  Any constraints on that?

**Steps to reproduce the issue:**
1.  Deploy Swarmpit 1.3 to a swarm 
2.  Login to swarmpit as admin.
2.  Created a shared developer role/account (user access level) in Swarmpit.
3.  Deploy a service using an image from a private dockerhub repo. (deployed from outside Swarmpit)
4.  Login to swarmpit as developer role/account.
5.  Attempt to edit that service.  

**What happens:**
It kicks back the login screen.  Looking at dev tools, it looks like there's a http 401 on the request of the url '/distribution/public/tags?repository=ourcompany/our-service'.   

**What should happen:**
It should allow editing the service.

**Additional information (e.g. docker version, cluster setup,...):**
docker version: 17.09.1-ce

**Comment by nohaapav** [comment]
@mpsmith Hi, i will push fix tomorrow. We did kind a huge refactoring in 1.3 on router and swarmpit 401 & 403 ex are redirecting automatically to custom page. Anyway redirect should be skipped if client exception is thrown (in this case registry client). 

Tags in this case cannot be prefetched because service is not created via swarmpit (unknown distribution). This will be improved once https://github.com/swarmpit/swarmpit/issues/197 is ready.
Should be this week i hope :)

**Comment by nohaapav** [comment]
@mpsmith Could you check now .. it should be fixed! .. Just pull new 1.3 image ;)

**Comment by johnjelinek** [comment]
![image](https://user-images.githubusercontent.com/873610/35111960-388d0d50-fc42-11e7-8d24-ddea595ec9b9.png)
It doesn't redirect you to the login, but it still doesn't allow me to edit services based on private images.

**Comment by nohaapav** [comment]
@johnjelinek strange, works for me. Did you clean browser cache ? 

![image](https://user-images.githubusercontent.com/4179104/35113576-31929d86-fc82-11e7-82a6-cd5674ea3d43.png)

**Comment by johnjelinek** [comment]
I was running in an incognito window when I took that gif. So cache has been cleared.

**Comment by nohaapav** [comment]
As i mentioned in previous comment could you check if GET `/services/{id}` and GET `/services/{id}/networks` end up with `200` ? 

It looks like some BE query failed in your case.

**Comment by nohaapav** [comment]
Should be fixed in https://github.com/swarmpit/swarmpit/releases/tag/1.3.1

**Comment by mpsmith** [comment]
@nohaapav Just tried 1.3.1    It allows me into the edit page of the service, but when you change something (i.e. - replicas), and click save, the following error pops on the bottom of the browser: "Service update failed. Reason: authentication required".      The POST to /services/xyz is returning a 401.


cc: @johnjelinek

**Comment by mpsmith** [comment]
Do you need anything else from us to help reproduce this?   Can we re-open the issue?

**Comment by mpsmith** [comment]
Or we can open a new issue if that is preferred (?)

**Comment by masterxavierfox** [comment]
Im having the same issue, when i try to edit a service that is from a private repo on docker hub specifically the replica count i get the following error "Service update failed. Reason: authentication required"

**Comment by mpsmith** [comment]
I created a new issue for the saving of the service here:  https://github.com/swarmpit/swarmpit/issues/203

**Comment by nohaapav** [comment]
@mpsmith I answered in new issue. Next time please mention directly issue assignee with @ so i can answer in no time ;)
